File: ios/Classes/LiquidGlassBridgePlugin.swift
import Flutter
import UIKit

public class LiquidGlassBridgePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let factory = LiquidGlassNativeViewFactory(messenger: registrar.messenger())
    registrar.register(factory, withId: "liquid_glass_bridge/native_glass_view")

    let channel = FlutterMethodChannel(
      name: "liquid_glass_bridge/native_glass",
      binaryMessenger: registrar.messenger()
    )
    let instance = LiquidGlassBridgePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "isRegistered":
      result(true)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/drraspec/liquid_glass_bridge/LiquidGlassBridgePlugin.kt
package com.drraspec.liquid_glass_bridge

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class LiquidGlassBridgePlugin : FlutterPlugin, MethodChannel.MethodCallHandler {
  private var channel: MethodChannel? = null

  override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    val factory = LiquidGlassNativeViewFactory()
    binding.platformViewRegistry.registerViewFactory(
      "liquid_glass_bridge/native_glass_view",
      factory
    )
    channel = MethodChannel(binding.binaryMessenger, "liquid_glass_bridge/native_glass")
    channel?.setMethodCallHandler(this)
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel?.setMethodCallHandler(null)
    channel = null
  }

  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
    when (call.method) {
      "isRegistered" -> result.success(true)
      else -> result.notImplemented()
    }
  }
}
